tests/unit/test_algebraic_expression.cpp

```diff
@@ -138,23 +138,18 @@ class AlgebraicExpressionTest: public ::testing::Test {
         int labels[11] = {person_label, person_label, person_label, person_label, person_label, person_label,
                         country_label, country_label, country_label, country_label, country_label};
 
-        DataBlockIterator *it;
-        Graph_CreateNodes(g, node_count, labels ,&it);
-
-        // Assign attributes to nodes.
-        char *default_property_name = (char *)"name";
-        for(int i = 0; i < person_count; i++) {
-            Node *n = (Node*)DataBlockIterator_Next(it);
-            SIValue name = SI_StringVal(persons[i]);
-            Node_Add_Properties(n, 1, &default_property_name, &name);
-        }
-        for(int i = 0; i < country_count; i++) {
-            Node *n = (Node*)DataBlockIterator_Next(it);
-            SIValue name = SI_StringVal(countries[i]);
-            Node_Add_Properties(n, 1, &default_property_name, &name);
-        }
+        char *default_property_name = (char *)"name";        
+        Graph_AllocateNodes(g, node_count);
 
-        DataBlockIterator_Free(it);
+        for(int i = 0; i<node_count; i++) {
+            Node n;
+            SIValue name;
+            Graph_CreateNode(g, labels[i], &n);
+
+            if(i < person_count) name = SI_StringVal(persons[i]);                
+            else name = SI_StringVal(countries[i]);
```

Comment:
Tries to access countries[person_count..node_count], which will crash. Can do `i - person_count` or solve some other way.

Reply:
👍